updater.js

```diff
@@ -0,0 +1,252 @@
+      })
+    })
+  })
+}
+
+function getLockfile () {
+  var packagePath = pkgUp.sync()
+  var rootDir = path.dirname(packagePath)
+
+  var lockfileNpm = path.join(rootDir, 'package-lock.json')
+  var lockfileYarn = path.join(rootDir, 'yarn.lock')
+  var lockfilePnpm = path.join(rootDir, 'pnpm-lock.yaml')
+
+  if (fs.existsSync(lockfileNpm)) {
+    packageManager = 'npm'
+    return lockfileNpm
+  }
+
+  if (fs.existsSync(lockfileYarn)) {
+    packageManager = 'yarn'
+    return lockfileYarn
+  }
+
+  if (fs.existsSync(lockfilePnpm)) {
+    packageManager = 'pnpm'
+    return lockfilePnpm
+  }
+
+  throw new BrowserslistError('Lockfile search error')
+}
+
+function getPackageManager () {
+  if (packageManager === 'yarn') {
+    return 'yarn'
+  }
+
+  if (packageManager === 'pnpm') {
+    return 'pnpm'
+  }
+
+  return 'npm'
+}
+
+function getCurrentVersion (lockfileRaw) {
+  if (packageManager === 'yarn') {
+    return getYarnCaniuseVersion(lockfileRaw)
+  }
+
+  if (packageManager === 'pnpm') {
+    return getPnpmCaniuseVersion(lockfileRaw)
+  }
+
+  return getNpmCaniuseVersion(lockfileRaw)
+}
+
+function getNpmCaniuseVersion (lockfileRaw) {
+  var parsedFile = JSON.parse(lockfileRaw)
+
+  if (
+    parsedFile.dependencies !== undefined &&
+    parsedFile.dependencies[PACKAGE_CANIUSE] !== undefined &&
+    parsedFile.dependencies[PACKAGE_CANIUSE].version !== undefined
+  ) {
+    return parsedFile.dependencies[PACKAGE_CANIUSE].version
+  }
+
+  return UNKNOWN_VERSION
+}
+
+function getYarnCaniuseVersion (lockfileRaw) {
+  var parsedFile = lockfileRaw.split('\n')
+  var foundCaniuse = false
+
+  for (var i = 0; i < parsedFile.length; i++) {
+    var line = parsedFile[i]
+
+    if (line.indexOf(PACKAGE_CANIUSE) === 0) {
+      foundCaniuse = true
+      continue
+    }
+
+    if (foundCaniuse && /version/.test(line)) {
+      var rule = /version "([^"]+)"/.exec(line)
+
+      if (rule.length > 1) {
+        return rule[1]
+      }
+
+      return UNKNOWN_VERSION
+    }
+
+    if (foundCaniuse && line === '') {
+      foundCaniuse = false
+    }
+  }
+
+  return UNKNOWN_VERSION
+}
+
+function getPnpmCaniuseVersion (lockfileRaw) {
+  var parsedFile = lockfileRaw.split('\n')
+
+  for (var i = 0; i < parsedFile.length; i++) {
+    var line = parsedFile[i]
+
+    if (line.indexOf('/' + PACKAGE_CANIUSE) >= 0) {
+      var rule = /\/([\d.]+):/.exec(line)
+
+      if (rule.length > 1) {
+        return rule[1]
+      }
+
+      return UNKNOWN_VERSION
+    }
+  }
+
+  return UNKNOWN_VERSION
+}
+
+function getLastVersion () {
+  var versions = childProcess
+    .execSync('npm show ' + PACKAGE_CANIUSE + ' version')
+    .toString()
+    .split('\n')
+
+  if (versions.length === 0) {
+    throw new BrowserslistError(
+      'Error getting the latest version of ' + PACKAGE_CANIUSE
+    )
+  }
+
+  return versions[0]
+}
+
+function updateLockfile (lockfileRaw) {
+  if (packageManager === 'yarn') {
+    return cleanupYarnLock(lockfileRaw)
```

Comment:
If you use `updateLockfile` name, you should use `updateYarnLock`, etc

Reply:
I think it will be better to keep all code in one function in `if (packageManager === 'yarn') {` without functions for each package managers